### PR TITLE
LPS-171225 Automatically register ConfigurationVisibilityControllers for each Feature Flag defined in the portal.properties

### PR DESCRIPTION
**Pull request by drewbrokke**
This is an update for [LPS-171225](https://issues.liferay.com/browse/LPS-171225).

### CODE REVIEW

**Comment by drewbrokke** [comment]
ci:test:relevant

**Comment by drewbrokke** [comment]
ci:test:sf

**Comment by melodywu** [comment]
ci:test:relevant

**Comment by melodywu** [comment]
Hello @,Please take a look into the relevant failure. Thanks. The test all failed and the error is：
```
[exec] org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':apps:cookies:cookies-impl:jar'.
     [exec] 	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.lambda$executeIfValid$3(ExecuteActionsTaskExecuter.java:186)
     [exec] 	at org.gradle.internal.Try$Failure.ifSuccessfulOrElse(Try.java:268)
     [exec] 	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeIfValid(ExecuteActionsTaskExecuter.java:184)
     [exec] 	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:173)
     [exec] 	at org.gradle.api.internal.tasks.execution.CleanupStaleOutputsExecuter.execute(CleanupStaleOutputsExecuter.java:109)
     [exec] 	at org.gradle.api.internal.tasks.execution.FinalizePropertiesTaskExecuter.execute(FinalizePropertiesTaskExecuter.java:46)
     [exec] 	at org.gradle.api.internal.tasks.execution.ResolveTaskExecutionModeExecuter.execute(ResolveTaskExecutionModeExecuter.java:62)
     [exec] 	at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:57)
     [exec] 	at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:56)
     [exec] 	at org.gradle.api.internal.tasks.execution.CatchExceptionTaskExecuter.execute(CatchExceptionTaskExecuter.java:36)
     [exec] 	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.executeTask(EventFiringTaskExecuter.java:77)
     [exec] 	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:55)
     [exec] 	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:52)
     [exec] 	at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:200)
     [exec] 	at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:195)
     [exec] 	at org.gradle.internal.operations.DefaultBuildOperationRunner$3.execute(DefaultBuildOperationRunner.java:75)
     [exec] 	at org.gradle.internal.operations.DefaultBuildOperationRunner$3.execute(DefaultBuildOperationRunner.java:68)
```

File: modules/apps/feature-flag/feature-flag-web/src/main/java/com/liferay/feature/flag/web/internal/company/feature/flags/CompanyFeatureFlagsFactory.java [hunk 1]
```diff
@@ -47,14 +47,11 @@ public CompanyFeatureFlags create(long companyId) {
...
 
-			if (!GetterUtil.getBoolean(
-					PropsUtil.get(FeatureFlagConstants.getKey("LPS-167698")))) {
-
-				return new CompanyFeatureFlags(Collections.emptyMap());
-			}
-
 			Map<String, FeatureFlag> featureFlagsMap = new HashMap<>();
 
+			boolean uIEnabled = GetterUtil.getBoolean(
```
[the remark this answers is not in the document]
Comment (drewbrokke) [reply modules/apps/feature-flag/feature-flag-web/src/main/java/com/liferay/feature/flag/web/internal/company/feature/flags/CompanyFeatureFlagsFactory.java:52]: I'll give it a shot. This was SF's suggestion and I didn't think to question it, except that it's very ugly 🤣 

I will be very happy if your suggestion works 🤞

[the remark this answers is not in the document]
Comment (drewbrokke) [reply modules/apps/feature-flag/feature-flag-web/src/main/java/com/liferay/feature/flag/web/internal/company/feature/flags/CompanyFeatureFlagsFactory.java:52]: Works! I'll update the PR. Thanks Pei!

**Comment by drewbrokke** [comment]
ci:test:relevant

**Comment by drewbrokke** [comment]
ci:test:sf

**Comment by melodywu** [comment]
No related issue found in ci:test:relevant.

**Comment by patriciajeanperez** [comment]
Hi @drewbrokke, no more related failures, just a version increase required for the following:
```
     [exec] com.liferay.semantic.versioning.SemanticVersioningTest > testSemanticVersioning[/apps/cookies/cookies-api] FAILED
     [exec]     java.lang.AssertionError:   PACKAGE_NAME                                       DELTA      CUR_VER    BASE_VER   REC_VER    WARNINGS  
     [exec]     = ================================================== ========== ========== ========== ========== ==========
     [exec]     * com.liferay.cookies.configuration.banner           CHANGED    1.0.0      1.0.0      1.0.1      VERSION INCREASE REQUIRED
     [exec]     	~   interface  com.liferay.cookies.configuration.banner.CookiesBannerConfiguration
     [exec]     		~   annotated  com.liferay.portal.configuration.metatype.annotations.ExtendedObjectClassDefinition
     [exec]     			+   property   visibilityControllerKey='com.liferay.cookies.configuration'
     [exec]     * com.liferay.cookies.configuration.consent          CHANGED    1.0.0      1.0.0      1.0.1      VERSION INCREASE REQUIRED
     [exec]     	~   interface  com.liferay.cookies.configuration.consent.CookiesConsentConfiguration
     [exec]     		~   annotated  com.liferay.portal.configuration.metatype.annotations.ExtendedObjectClassDefinition
     [exec]     			+   property   visibilityControllerKey='com.liferay.cookies.configuration'
```

Thank you!

**Comment by drewbrokke** [comment]
ci:test:relevant

**Comment by drewbrokke** [comment]
@patriciajeanperez are these failures related?

**Comment by patriciajeanperez** [comment]
No related failures @drewbrokke

**Pull request closed without merging**